src/borg/archiver.py

```diff
@@ -533,6 +533,39 @@ def create_inner(archive, cache, fso):
                 else:
                     status = '-'
                 self.print_file_status(status, path)
+            elif args.paths_from_command or args.paths_from_stdin:
+                paths_sep = '\n'
+                if args.paths_delimiter is not None:
+                    # evaluate escape sequences in the separator, in case the shell doesn't replace eg '\\0'
+                    paths_sep = escape_decode(args.paths_delimiter.encode('utf-8'))[0].decode('utf-8')
```

Comment:
maybe a helper function for this would be nice, guess we might also need that at other places.

when done, these 4 lines could be like:
```
paths_sep = thathelper(args.path_delimiter) if args.path_delimiter is not None else '\n'
```

Reply:
I put it in `helpers/parseformat.py`, seems like there is a bunch of encoding stuff there already. Btw i changed the escape-evaluation mechanism again. `codecs.escape_decode` wasn't really public api (some old python2 leftover apparently) and now i understand `unicode_escape` better.

Reply:
ok, great!